docs(chatbot): add component pages (#244)
siemens/ix-docs: docs/components/chat-attachment/guide.md

---
doc-type: 'tab-item'
description: 'Usage guide for AI message'
---

# Chat attachment - Usage

Used to display a single response generated by the AI in a conversational chat thread

> content will come here

Reviewer: When will that content come?

Reviewer 2: added ✅ I'll leave your comment open to re-review

---
doc-type: 'tab-item'

> description: 'Usage guide for chat attachment'

Reviewer 3: full stop?

---

# Chat attachment - Usage

> Chat attachments display files that users add to a chat prompt. We recommend using them in [chat inputs](../chat-input) while users compose a message and in [user messages](../user-message) after submission to preserve the full request context.

Reviewer 3: ```suggestion
Chat attachments display files that users have uploaded to a chat prompt. It is typically displayed when users are writing in the chat input [chat inputs](../chat-input) and in all sent user messages.
```

![Chat attachment anatomy](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7970-230&t=HrpSIFfB7yjzt741-4)

1. File icon
2. File name
3. Remove button

## Options

- **File name:** Show the file name including the file extension.
- **Remove button:** Show remove buttons in a [chat input](../chat-input/), and hide it once an attachment is part of a submitted [message](../user-message/).

> - **Preview supported:** Use only for supported files to show e.g. thumbnails.

Reviewer 3: is this part of the UX AI guidelines?

Reviewer 2: yes, https://www.figma.com/design/KbgPxj7qLgngXkJfnDM4Ty/SDL-AI-UX-Guidelines?node-id=11900-151929&t=pbv0c2rQagXjKN4E-11

- **Tooltip:** By default, the tooltip shows the file name in case of truncation. Override it to show additional information, e.g. for previews or file size.

## Behavior in context

- **Chat input context:** If attachments exceed the chat input's width, they overflow into a scrollable horizontal list.
- **User message context:** If attachments exceed the user message container's width, a more button is visible.
- **Text overflow:** If an attachment exceeds `20rem`, the file name is truncated at the end while preserving the file extension. A tooltip shows the full file name on hover.

## States

Chat attachments have six states: default, hover, active, loading, focused and error.

Chat attachments follow the [chip](../chip) interaction model, including hover, active and focused behavior. In an error state, attachments stay visible and show clear feedback so users can retry or remove files.

## Dos and Don’ts

> - Do keep submitted attachments visible in user messages so conversation context stays traceable

Reviewer 3: ```suggestion
- Do keep attachments visible in user messages so attachments stay traceable within context
```

- Don’t hide the remove action while users are still composing a prompt in the [chat input](../chat-input/)

> - Don’t detach attachments from their related [user message](../user-message/) after submission

Reviewer 3: ```suggestion
- Don’t detach attachments from their related [user message](../user-message/) after sending 
- 
```

Note that the original action verb was submit, but now send

---
doc-type: 'tab-item'
description: 'Usage guide for chat attachment.'
---

# Chat attachment - Usage

Chat attachments display files that users have uploaded to a chat prompt. It is typically displayed when users are writing in the chat input [chat inputs](../chat-input) and in all sent [user messages](../user-message/).

![Chat attachment anatomy](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7970-230&t=HrpSIFfB7yjzt741-4)

1. File icon
2. File name
3. Remove button

## Options

> - **File icon:** Show a recognizable file-type icon so users can identify attachments quickly, e.g. [pdf-document](../../icons/icon-library).

Reviewer: Either link "icon" in the first part of the sentence to our icon library in general or provide a direct link to the pdf-document icon here. 

Nice side effect: with the new icon search, ppt and doc are shown in the related icons
https://ix.siemens.io/docs/icons/icon-library#pdf-document

Reviewer 2: I tried to do a direct link with the # but the pipeline failed. Daniel said that it works differently because they are built differently than normal content. I'll change it to link the first part!

- **File name:** Show the file name including the file extension.

> - **Remove button:** Show remove buttons in a [chat input](../chat-input/), and hide it once an attachment is part of a submitted [message](../user-message/).

Reviewer: omit the comma before "and"

- **Preview supported:** Use only for supported files to show e.g. thumbnails.

## Behavior in context

- **Chat input context:** If attachments exceed the chat input's width, they overflow into a scrollable horizontal list.
- **User message context:** If attachments exceed the user message container's width, a more button is visible.
- **Text overflow:** If an attachment exceeds `20rem`, the file name is truncated at the end while preserving the file extension.

## States

Chat attachments have six states: default, hover, active, loading, focused and error.

Chat attachments follow the [chip](../chip) interaction model, including hover, active and focused behavior. In an error state, attachments stay visible and show clear feedback so users can retry or remove files.

## Dos and Don’ts

- Do keep attachments visible in user messages so attachments stay traceable within context
- Don’t hide the remove action while users are still composing a prompt in the [chat input](../chat-input/)
- Don’t detach attachments from their related [user message](../user-message/) after sending

> -

Reviewer: a very lonely bullet point

---
doc-type: 'tab-item'
description: 'Usage guide for chat attachment'
---

# Chat attachment - Usage

Chat attachments display files that users add to a chat prompt. We recommend using them in [chat inputs](../chat-input) while users compose a message and in [user messages](../user-message) after submission to preserve the full request context.

![Chat attachment anatomy](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7970-230&t=HrpSIFfB7yjzt741-4)

1. File icon
2. File name
3. Remove button

## Options

- **File icon:** Show a recognizable file-type icon so users can identify attachments quickly, e.g. pdf-document.
- **File name:** Show the file name including the file extension.
- **Remove button:** Show remove buttons in a [chat input](../chat-input/), and hide it once an attachment is part of a submitted [message](../user-message/).

> - **Preview supported:** Use only for supported files to show e.g. thumbnails.

Reviewer: Would turn it around to first say what it does (showing file preview, e.g. thumbnails) and then "only if supported"

## Behavior in context

- **Chat input context:** If attachments exceed the chat input's width, they overflow into a scrollable horizontal list.
- **User message context:** If attachments exceed the user message container's width, a more button is visible.

> - **Text overflow:** If an attachment exceeds `20rem`, the file name is truncated at the end while preserving the file extension.

Reviewer: do we offer a solution or could at least recommend one, e.g. hovering shows the file name in a tooltip?

Reviewer 2: @danielleroux ?

Reviewer 2: added tooltip description ✔️

Reviewer: @kathrinschalber Somehow I cannot see the tooltip description, can you check?

Reviewer 2: true, somehow it got removed. Maybe I forgot to actually push it 😂

## States

Chat attachments have six states: default, hover, active, loading, focused and error.

Chat attachments follow the [chip](../chip) interaction model, including hover, active and focused behavior. In an error state, attachments stay visible and show clear feedback so users can retry or remove files.

> ## Dos and Don’ts

Reviewer: are these dos and don'ts possible to deviate from or is this behavior baked into the component?

Reviewer 2: there's the option to add attachments to our recommended predefined position, but they could leave it out there (just not adding them there), and place them somewhere else 🤔
